src/metabase/models/serialization.clj

```diff
@@ -844,10 +844,20 @@
       (cond
         (nil? entity)      (throw (ex-info "FK target not found" {:model model
                                                                   :id    id
-                                                                  :skip  true}))
+                                                                  :skip  true
+                                                                  ::type :target-not-found}))
         (= (count path) 1) (first path)
         :else              path))))
 
+(defn- export-fk?
```

Comment:
Take it or leave it, but I'm not sure about naming this `export-fk?` when it returns nil or a fk (or a seq of fks). [The style guide calls them predicate methods](https://github.com/bbatsov/clojure-style-guide?tab=readme-ov-file#predicate-methods). 

Seems like it's being used as a predicate (with the `when` below), but also not, so maybe it's ok. I can't think of a great way to express it... Maybe introducing another function that is truly a predicate so that they both have a clear use-case rather than: We use this to either check if it's truthy/falsey, _or_ use the fk returned value?